### DISCUSSION TITLE
Proposal for retrieving workflow executions by correlationId across all workflows

### DISCUSSION THREAD

**Discussion by raedbh** [Netflix/conductor#3527] [Ideas]
As we're currently using Conductor with only MySQL as persistence module (without ES), we're in need of a way to retrieve the list of workflow executions by correlationId without having to specify the workflow name. To address this, I propose to add the following method to the `WorkflowService` and implement it:

```
    /**
     * Lists workflow executions for all workflows with the given correlation id.
     *
     * @param correlationId CorrelationID of the workflows you want to list.
     * @param includeClosed IncludeClosed workflow which are not running.
     * @param includeTasks Includes tasks associated with workflows.
     * @return a list of {@link Workflow}
     */
    List<Workflow> getWorkflows(
            String correlationId,
            boolean includeClosed,
            boolean includeTasks);
```

This will enable us to retrieve the relevant workflow executions across all workflows with the same correlationId (even when we don't know the specific workflow names). 
I welcome your thoughts and feedback on this proposal.

**Comment by v1r3n** [comment]
Hi @raedbh We have this in the pipeline.  Here is the API we are looking at supporting:

```
curl -X 'POST' \
  'https://play.orkes.io/api/workflow/correlated/batch?includeClosed=false&includeTasks=false' \
  -d '{
  "correlationIds": [],
  "workflowNames": []
}'
```

This will allow sending a batch of correlationIds and workflow names (which can be empty) to find workflows matching any from the list of correlationIds and workflow names.

**Reply by raedbh** [reply]
Hi Viren, thanks for your reply, looking forward to see this API in place ^^
